toqito/matrices/tests/test_gen_pauli_z.py

```diff
@@ -0,0 +1,139 @@
+"""Test gen_pauli_z."""
+
+import numpy as np
+import pytest
+
+from toqito.matrices import gen_pauli_z
+
+
+@pytest.mark.parametrize("dim", range(2, 1024, 64))
+def test_shape(dim):
+    """Ensure the shape of a generalized Pauli Z is what we expect."""
+    assert gen_pauli_z(dim).shape == (dim, dim)
+
+
+@pytest.mark.parametrize(
+    "dim, want",
+    [
+        (1, np.array([[1.0]])),
+        (
+            3,
+            np.array(
+                [
+                    [1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
+                    [0.0 + 0.0j, -0.5 + 0.8660254j, 0.0 + 0.0j],
+                    [0.0 + 0.0j, 0.0 + 0.0j, -0.5 - 0.8660254j],
+                ]
+            ),
+        ),
+        (
+            9,
+            np.array(
```

Comment:
For the `dim=9` case the line lengths are over 120 chars. We could delete the complex zeros, and use something like the following?

```
+                    [1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.76604444 + 0.64278761j, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.17364818 + 0.98480775j, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, -0.5 + 0.8660254j, 0.0, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, -0.93969262 + 0.34202014j, 0.0, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, 0.0, -0.93969262 - 0.34202014j, 0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -0.5 - 0.8660254j, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.17364818 - 0.98480775j, 0.0],
+                    [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.76604444 - 0.64278761j],
```

Reply:
Could you maybe use `dim<9`? It could help with the formatting temporarily. 

I'll work on updating the configuration settings for `ruff`. https://github.com/vprusso/toqito/issues/512#issuecomment-2073965485

Reply:
The suggestion above works with the style constraints and still gives dtype=complex128 over the whole array. What do you think @purva-thakre ?